File: python/unidad_03/problema_3_03.py
'''
El programa calcula el resultado de una determinada serie.
'''

def calcular_serie(numero, serie):

    var_control = True

    for i in range (1, numero + 1):

        if var_control:

            serie += 1 / i
            var_control = False

            print(f"{serie:.4f}")
            continue
        
        serie -= 1 / i
        var_control = True

        print(f"{serie:.4f}")
    
    return serie

def main():

    serie = 0

    try:

        numero = int(input("Número entero: "))

    except ValueError as e:

        print(e)
        return
    
    serie = calcular_serie(numero, serie)

    print(f"Resultado de la serie: {serie:.4f}")

if __name__ == '__main__':

    main()

File: java/unidad_03/Problema_3_03/src/Problema_3_03.java
import java.util.InputMismatchException;
import java.util.Scanner;

/*
 * El programa calcula el resultado de una determinada serie.
 */

public class Problema_3_03 {

    public static float calcularSerie(int numero, float serie) {

        boolean varControl = true;

        for (int i = 1; i <= numero; i++) {

            if (varControl) {

                serie += 1.0 / i;
                varControl = false;

                System.out.printf("%.4f %n", serie);
                continue;
            }
            serie -= 1.0 / i;
            varControl = true;

            System.out.printf("%.4f %n", serie);
        }
        return serie;
    }

    public static void main(String[] args) {

        int numero;

        float serie = 0;

        try (Scanner scanner = new Scanner(System.in)) {

            System.out.print("Número entero: ");
            numero = scanner.nextInt();

        } catch (InputMismatchException e) {

            System.out.println(e);
            return;
        }
        serie = calcularSerie(numero, serie);

        System.out.printf("Resultado de la serie: %.4f", serie);
    }
}